pkgs/development/libraries/valhalla/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib, stdenv, fetchFromGitHub, fetchpatch, cmake, pkgconfig
+, zlib, curl, protobuf, prime-server, boost, sqlite, libspatialite, lua, geos
+, python3, python3Packages, rapidjson, zeromq }:
+
+let
+  osmbinary = fetchFromGitHub {
+    owner = "openstreetmap";
+    repo = "OSM-binary";
+    rev = "4e32fa236e919be5bb8dbb323437f026f24e5853";
+    sha256 = "1ylxbhli447j4i606cpi9cahym173fblnm3sgwj372wvxf9yz9yv";
+  };
+  date = fetchFromGitHub {
+    owner = "HowardHinnant";
+    repo = "date";
+    rev = "3e82a52d665cdaaa51027f15906bf503b6896142";
+    sha256 = "0czmj0y5fw92n1dncad6lc42v4kc9rvkz2703410wxyaphrlc4sg";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "valhalla";
+  version = "3.0.9";
+
+  src = fetchFromGitHub {
+    owner = "valhalla";
+    repo = "valhalla";
+    rev = version;
+    sha256 = "0y2d75w0maag5zcv4w0rnprcsn0k2myzhqs1fbgwm60s42kjghf8";
+  };
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [
+    zlib curl protobuf prime-server boost sqlite libspatialite lua geos
+    python3 python3Packages.boost rapidjson zeromq
+  ];
+
+  postUnpack = ''
+    cp -vr ${osmbinary}/* source/third_party/OSM-binary/
+    cp -vr ${date}/* source/third_party/date/
+    chmod -R +w source/third_party
+  '';
+
+  patches = [
+    # Can't use fetchpatch as upstream commit also updates changelog, causing conflict:
```

Comment:
fetchpatch can ignore files. Please use that feature instead.

fetchpatch {
  url = "https://github.com/valhalla/valhalla/commit/0538e2562e858896e517d1a3e09e8158f697e184.patch";
  sha256 = "...";
  excludes = "CHANGELOG.md";
}

Reply:
I never knew you could do that. Probably because it isn't mentioned in the documentation :unamused:.